core/src/main/java/org/elasticsearch/action/support/AutoCreateIndex.java

```diff
@@ -87,7 +88,15 @@ public boolean shouldAutoCreate(String index, ClusterState state) {
         return false;
     }
 
```

Comment:
can you update the `shouldAutoCreate` method so that it getl a local reference to the AutoCreate object as a first step and then uses this reference for checks? The motivation is to make sure all checks are done against the same instance in case there are concurrent updates of the auto_create setting